src/ngResource/resource.js

```diff
@@ -640,7 +642,7 @@ angular.module('ngResource', ['ng']).
         };
 
         forEach(actions, function(action, name) {
-          var hasBody = /^(POST|PUT|PATCH)$/i.test(action.method);
+          var hasBody = /^(POST|PUT|PATCH)$/i.test(action.method) || action.hasBody === true;
```

Comment:
I had to like twice to see the error but the correct statement would be:
`var hasBody = action.hasBody === true || (action.hasBody === false && /^(POST|PUT|PATCH)$/i.test(action.method));`